src/components/SpreadsheetUploader/index.js

```diff
@@ -0,0 +1,189 @@
+        return;
+      }
+
+      setData(uploadedData);
+    };
+
+    reader.onerror = function (event) {
+      console.error('File could not be read! Code ' + event.target.error.code);
+    };
+
+    reader.readAsBinaryString(file);
+  };
+
+  const dropZone = (
+    <>
+      <Dropzone onDropAccepted={(files) => onDropAccepted(files)} multiple={false}>
+        {({ getRootProps, getInputProps }) => {
+          return (
+            <section>
+              <div className={styles.dropzone} {...getRootProps()}>
+                <input {...getInputProps()} />
+                <img alt={'Spreadsheet Icon'} src={SpreadsheetSVG} style={{ width: 40 }} />
+                Upload a Spreadsheet
+              </div>
+            </section>
+          );
+        }}
+      </Dropzone>
+      <Typography style={{ textAlign: 'center', fontSize: 10 }} variant="body2">
+        Accepted file types: CSV, XLSX
+      </Typography>
+      {template ? (
+        <Link
+          to={template}
+          target="_blank"
+          download
+          style={{ color: gordonColors.primary.blue, textAlign: 'center' }}
+        >
+          Download Template
+        </Link>
+      ) : null}
+    </>
+  );
+
+  useEffect(() => {
+    if (error) {
+      setData(null);
+    }
+  }, [error]);
+
+  return (
+    <>
+      <GordonDialogBox
+        open={open}
+        title={title}
+        cancelButtonClicked={() => {
+          setOpen(false);
+        }}
+      >
+        {data ? (
+          <>
+            <div style={{ height: '5px' }}></div>
+            <Button
+              variant="contained"
+              onClick={() => {
+                onSubmitData(data);
+                setOpen(false);
+                setData(null);
+              }}
+              color="primary"
+              disabled={!data}
+            >
+              {buttonName}
+            </Button>
+            <Button
+              variant="contained"
+              onClick={() => {
+                setOpen(false);
+                setData(null);
+              }}
+            >
+              Cancel
+            </Button>
+            <div style={{ height: '15px' }}></div>
+            <Typography
+              variant="body2"
+              width="60%"
+              style={{
+                marginTop: '100px',
+                width: '60%',
+                margin: 'auto',
+                textAlign: 'center',
+                fontWeight: 'bold',
+              }}
+            >
+              Check over the data below to confirm that it is accurate.
+            </Typography>
+            <div style={{ height: '10px' }}></div>
+          </>
+        ) : null}
+        {data
+          ? data.map((row, index) => {
+              const fields = requiredColumns.map((c) => {
+                return row[c] ? (
+                  <Typography variant="body2">
+                    <b>{c}:</b> {row[c]}
+                  </Typography>
+                ) : null;
+              });
```

Comment:
I feel like sometimes this hurts code readability when it is a multi-line expression with JSX in the return. That is why I added brackets, not because I don't know about expression syntax.

Reply:
Nevertheless, I am making changes as you requested